### PR TITLE
BDD tests - services check

### PR DESCRIPTION
**Pull request by tisnik**
# Description

BDD tests - services check

## Type of change

- Behavioral tests

## Testing steps

N/A

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: bdd_tests/features/cli_flags.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     Then I should see version info displayed on standard output
+
+
+  Scenario: Check if CCX Notification Writer displays authors
+    Given the system is in default state
+     When I start the CCX Notification Writer with the --authors command line flag
+     Then I should see info about authors displayed on standard output
+
+
+  Scenario: Check the ability to drop initialize all database tables
```
Comment (epapbak) [inline bdd_tests/features/cli_flags.feature:22]: remove 'drop'

File: bdd_tests/features/cli_flags.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    Given Postgres is running
+      And CCX Notification Writer database is created for user postgres with password postgres
```
Comment (epapbak) [inline bdd_tests/features/cli_flags.feature:24]: I would add "And CCX Notification Writer database is not set up" condition, which would check that the tables do not exist in the DB.

File: bdd_tests/features/smoketests.feature [hunk 3]
```diff
@@ -7,21 +7,46 @@ Feature: Basic set of smoke tests
...
-    Given the system is in default state
-     When I connect to database named test as user postgres with password postgres
+    Given Postgres is running
+     When I connect to database named postgres as user postgres with password postgres
      Then I should be able to connect to such database
      When I close database connection
      Then I should be disconnected
+
+
+  Scenario: Check if CCX Notification Writer database can be reached
```
Comment (epapbak) [inline bdd_tests/features/smoketests.feature:30]: I don't see the value of this scenario

File: bdd_tests/features/smoketests.feature [hunk 4]
```diff
@@ -7,21 +7,46 @@ Feature: Basic set of smoke tests
...
+  Scenario: Check if Kafka broker is running locally
+    Given the system is in default state
+     When I retrieve a list of all applications running under JVM
+     Then I should find the following application kafka.Kafka
+
+
+  Scenario: Check if Kafka broker is running on expected port
+    Given the system is in default state
+     When I retrieve metadata from Kafka broker running on localhost:9092
+     Then I should found at least one available broker
```
Comment (epapbak) [inline bdd_tests/features/smoketests.feature:52]: find

**Review by epapbak — APPROVED** [review]
some small comments, but overall it's good to go

File: bdd_tests/features/cli_flags.feature (hunk 2 above)
Comment (tisnik) [reply bdd_tests/features/cli_flags.feature:24]: will do, thank you

File: bdd_tests/features/smoketests.feature (hunk 3 above)
Comment (tisnik) [reply bdd_tests/features/smoketests.feature:30]: it's just a smoke pre-test for other scenarios, but you are right - it actually do the check in "Given" clause :)

File: bdd_tests/features/cli_flags.feature (hunk 1 above)
Comment (tisnik) [reply bdd_tests/features/cli_flags.feature]: Fixed

File: bdd_tests/features/cli_flags.feature (hunk 2 above)
Comment (tisnik) [reply bdd_tests/features/cli_flags.feature:24]: Implemented via new commit

File: bdd_tests/features/smoketests.feature (hunk 4 above)
Comment (tisnik) [reply bdd_tests/features/smoketests.feature]: fixed

**Pull request merged**